### PR TITLE
Add headless service for StatefulSet

### PR DESCRIPTION
**Pull request by renesch-de**
In a Kubernetes StatefulSet, the serviceName field specifies the Headless Service associated with the StatefulSet. This service is responsible for network identity, including network routing and DNS naming, for the pods based on their ordinal index.

Also removed the redundant 'statefulset.name' and 'statefulset.serviceName' from README and values file as they're not used or required anymore.

### CODE REVIEW

**Review by bobotimi — COMMENTED** [review]
@renesch-de Thank you for your PR. Please can we make the creation of the headless service configurable and set to true by default. That is,

```
service:
  headless:
    enabled: true
    publishNotReadyAddresses: true

File: nxrm-ha/templates/service-headless.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+{{- end }}
+  labels:
+  {{- include "nexus.labels" . | nindent 4 }}
+  {{- if .Values.nexus.extraLabels }}
+    {{- with .Values.nexus.extraLabels }}
+      {{ toYaml . | nindent 4 }}
+    {{- end }}
+  {{- end }}
+spec:
+  type: ClusterIP
```
[the remark this answers is not in the document]
Comment (renesch-de) [reply nxrm-ha/templates/service-headless.yaml:18]: @bobotimi Agreed, it will default to this value anyway if not set.

**Comment by renesch-de** [comment]
@bobotimi On top of making `publishNotReadyAddresses` configurable, I added the option to define additional annotations for the headless service alone. The annotations are merged with the common service annotations already configurable via `service.annotations`

File: nxrm-ha/tests/service-headless_test.yaml [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            a2: annotation2
+  - it: should create headless service with publishNotReadyAddresses set to false
+    set:
+      service:
+        headless:
+          publishNotReadyAddresses: false
+    asserts:
+      - equal:
+          path: spec.publishNotReadyAddresses
+          value: false
```
Comment (bobotimi) [inline nxrm-ha/tests/service-headless_test.yaml:125]: I see we have tests for additional annotations and for replacing keys . We should have one for when `service.headless.annotations` is not specified?

**Review by bobotimi — COMMENTED** [review]
@renesch-de Thanks for your PR. Just one comment about an additional test

**Review by bobotimi — APPROVED** [review]

**Review by igorserdyuchenko — APPROVED** [review]

**Pull request merged**